base_business_document_import/models/business_document_import.py

```diff
@@ -109,31 +119,20 @@ def _match_partner(self, partner_dict, chatter_msg, partner_type="supplier"):
             )
             if state:
                 domain += ["|", ("state_id", "=", False), ("state_id", "=", state.id)]
+
+        # Append required vat to the domain
         if partner_dict.get("vat"):
-            vat = partner_dict["vat"].replace(" ", "").upper()
```

Comment:
@jbaudoux Why do you remove the search with 'vat' only as criteria?

Reply:
@lmignon I'd like to support complex partner structures. The customer could be a contact child of the company. So I removed first match on commercial entity only and replaced by adding the vat in the domain to restrict to the right partner structure on the following searches.